resources/lib/upgrade_controller.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+"""Check if the addon has been updated and make necessary changes"""
+from __future__ import unicode_literals
+
+import resources.lib.common as common
+
+from resources.lib.globals import g
+from resources.lib.database.db_update import run_local_db_updates, run_shared_db_updates
+
+
+def check_addon_upgrade():
+    """Check addon upgrade and perform necessary update operations"""
+    addon_previous_ver = g.LOCAL_DB.get_value('addon_previous_version', g.VERSION)
+    addon_current_ver = g.VERSION
+    if addon_current_ver != addon_previous_ver:
```

Comment:
as 'addon_previous_ver' was set to current version, they are never different, so 'addon_previous_ver' is never set in the DB, so update won't ever occur.

addon_previous_ver should be set to something in the default DB imo :)

Reply:
whooops 😓 i made a loop
fortunately you help me to verify

Reply:
I can fix it with 'None' as default value in the commit I have to make for library migration. I wish to use the add-on controller to perform that operation.

Reply:
Plus I'd like to import username and password from the xml file if some exists, to avoid entering credentials again, if you agree :)

Reply:
i have already included fix to the original commit

> Plus I'd like to import username and password from the xml file if some exists, to avoid entering credentials again, if you agree :)

aaah no is just a passage, it would become dead code within a release...
someone will be annoyed but will pass😂

Reply:
As you wish boss :D

Reply:
> i have already included fix to the original commit

What's the default value you used?